hubspot/tasks_test.py

```diff
@@ -79,3 +119,46 @@ def test_sync_line_item_with_hubspot(mock_hubspot_request):
     mock_hubspot_request.assert_called_once_with(
         "LINE_ITEM", HUBSPOT_SYNC_URL, "PUT", body=body
     )
+
+
+def test_sync_errors_first_run(mock_hubspot_errors, mock_logger):
+    """Test that HubspotErrorCheck is created on 1st run and nothing is logged"""
+    mock_hubspot_errors.return_value.json.side_effect = [
+        error_response_json,
+        {"results": []},
+    ]
+    assert HubspotErrorCheck.objects.count() == 0
+    check_hubspot_api_errors()
+    assert HubspotErrorCheck.objects.count() == 1
+    assert mock_hubspot_errors.call_count == 1
+    assert mock_logger.call_count == 0
+
+
+def test_sync_errors_new_errors(mock_hubspot_errors, mock_logger):
+    """Test that errors more recent than last checked_on date are logged"""
+    last_check = HubspotErrorCheckFactory.create(
+        checked_on=datetime(2015, 1, 1, tzinfo=pytz.utc)
```

Comment:
Are these datetimes directly linked to the values in the timestamps in the example json above? It would be really nice if those timestamps and these datetimes were keyed off of the same values. Just looking at this test case, it's very unclear as to why this `checked_on` datetime would result in 2 errors being logged.

Reply:
Yes, there are 2 errors in the example JSON, and since this datetime is before both of the timestamps for the errors, both errors will be logged.